PWGJE/Tasks/crCharmHadronization.cxx

```diff
@@ -0,0 +1,159 @@
+// All rights not expressly granted are reserved.
+//
+// This software is distributed under the terms of the GNU General Public
+// License v3 (GPL Version 3), copied verbatim in the file "COPYING".
+//
+// In applying this license CERN does not waive the privileges and immunities
+// granted to it by virtue of its status as an Intergovernmental Organization
+// or submit itself to any jurisdiction.
+
+// charm lambda/d0 ratio hadronization task
+//
+/// \author Christian Reckziegel <christian.reckziegel@cern.ch>
+//
+#include "TVector3.h"
+
+#include "fastjet/PseudoJet.hh"
+#include "fastjet/ClusterSequenceArea.hh"
+
+#include "CommonConstants/PhysicsConstants.h"
+#include "Framework/AnalysisTask.h"
+#include "Framework/AnalysisDataModel.h"
+#include "Framework/ASoA.h"
+#include "Framework/O2DatabasePDGPlugin.h"
+#include "Framework/HistogramRegistry.h"
+
+#include "Common/Core/TrackSelection.h"
+#include "Common/Core/TrackSelectionDefaults.h"
+#include "Common/DataModel/EventSelection.h"
+#include "Common/DataModel/TrackSelectionTables.h"
+
+#include "PWGHF/DataModel/CandidateReconstructionTables.h"
+#include "PWGHF/DataModel/CandidateSelectionTables.h"
+
+#include "PWGJE/DataModel/Jet.h"
+#include "PWGJE/DataModel/JetSubstructure.h"
+#include "PWGJE/Core/JetFinder.h"
+#include "PWGJE/Core/FastJetUtilities.h"
+#include "PWGJE/Core/JetHFUtilities.h"
+
+using namespace o2;
+using namespace o2::framework;
+using namespace o2::framework::expressions;
+
+// creating table for storing distance data
+namespace o2::aod
+{
+namespace DistanceSpace
+{
+DECLARE_SOA_COLUMN(JetHfDist, jethfdist, float);
+DECLARE_SOA_COLUMN(JetPt, jetpt, float);
+DECLARE_SOA_COLUMN(JetEta, jeteta, float);
+DECLARE_SOA_COLUMN(JetPhi, jetphi, float);
+DECLARE_SOA_COLUMN(HfPt, hfpt, float);
+DECLARE_SOA_COLUMN(HfEta, hfeta, float);
+DECLARE_SOA_COLUMN(HfPhi, hfphi, float);
+DECLARE_SOA_COLUMN(HfMass, hfmass, float);
+DECLARE_SOA_COLUMN(HfY, hfy, float);
+} // namespace DistanceSpace
+DECLARE_SOA_TABLE(JetDistanceTable, "AOD", "JETDISTTABLE",
+                  DistanceSpace::JetHfDist,
+                  DistanceSpace::JetPt,
+                  DistanceSpace::JetEta,
+                  DistanceSpace::JetPhi,
+                  DistanceSpace::HfPt,
+                  DistanceSpace::HfEta,
+                  DistanceSpace::HfPhi,
+                  DistanceSpace::HfMass,
+                  DistanceSpace::HfY);
+} // namespace o2::aod
+
+// NB: runDataProcessing.h must be included after customize!
+#include "Framework/runDataProcessing.h"
+
+struct crCharmHadronizationTask {
+  // producing new table
+  Produces<aod::JetDistanceTable> distJetTable;
+
+  // Histogram registry: an object to hold your histograms
+  HistogramRegistry registry{"histos", {}, OutputObjHandlingPolicy::AnalysisObject};
+
+  // histogram configurables
+  Configurable<int> distBins{"distBins", 1000, "number of bins in distance histogram"};
```

Comment:
does this need to be a configurable? probably fine to hard code it

Reply:
All right!